test/integration/component/test_acl_listvolume.py

```diff
@@ -340,8 +340,19 @@ def setUpClass(cls):
             cls.vm_a_volume = Volume.list(cls.apiclient, virtualmachineid=cls.vm_a.id)
 
             cls.cleanup = [
+                cls.vm_d1,
+                cls.vm_d1a,
+                cls.vm_d1b,
+                cls.vm_d11,
+                cls.vm_d11a,
+                cls.vm_d11b,
+                cls.vm_d111a,
+                cls.vm_d12a,
+                cls.vm_d12b,
+                cls.vm_d2,
+                cls.vm_a
                 cls.account_a,
-                cls.service_offering,
+                cls.service_offering
```

Comment:
better have these added directly after the respective creates, otherwise no cleaning will happen on exceptions.